File: water/src/main/java/com/app/water/base/LoginActivity.java
package com.app.water.base;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.app.water.MainActivity;
import com.app.water.R;
import com.example.lib.domain.Constant;
import com.example.lib.domain.LoginData;
import com.example.lib.http.OkHttpUtils;
import com.example.lib.http.response.GsonResponseHandler;
import com.example.lib.utils.CommonUtil;
import com.example.lib.utils.HttpUtil;

import java.util.HashMap;

public class LoginActivity extends AppCompatActivity {


    private void LoginSuccessHandle() {
        Log.i("test", "LoginSuccessHandle: ");
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        findViewById(R.id.btnLogin).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                StartValidateLoginThread();
            }
        });
    }



    /**
     * 启动验证登录线程，Android 3.0以后只允许异步访问网络
     */
    private void StartValidateLoginThread() {
        HashMap<String, String> params = new HashMap<>();
        params.put(Constant.PARAM_USERID,"zhangyuanhao");
        params.put(Constant.PARAM_PASSWORD,"123456");
        try {
            params.put(Constant.PARAM_PASSWORD, CommonUtil.getMD5Str("123456"));
        } catch (Exception e) {
            e.printStackTrace();
        }
        OkHttpUtils.getInstance().post(LoginActivity.this, HttpUtil.getServiceAddress(Constant.LOGIN),params , new GsonResponseHandler<LoginData>() {
            @Override
            public void onSuccess(int statusCode, LoginData data) {
                if(statusCode == 1){
                    Log.i("test", "onSuccess: 登录成功");
                    LoginSuccessHandle();
                }else{
                    Log.i("test", "onSuccess: 登录失败");
                }
            }
            @Override
            public void onFailure(int statusCode, String error_msg) {
                Log.i("test", "onSuccess: 网络失败");
            }
        });
    }


}

File: yun/src/main/java/com/app/yun/LoginActivity.kt
package com.app.yun

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.CheckBox
import android.widget.Toast
import com.app.myapplication.common.LogUtil.d
import com.app.myapplication.common.domain.Constant
import com.app.myapplication.common.domain.LoginData
import com.app.myapplication.common.showToast
import com.app.myapplication.common.utils.CommonUtil
import com.app.myapplication.common.utils.http.CookieHelper
import com.app.myapplication.common.utils.http.GsonResponseHandler
import com.app.myapplication.common.utils.http.HttpUtil
import com.app.myapplication.common.utils.http.OkHttpUtils
import com.app.myapplication.common.utils.ui.UIUtil
import com.app.myapplication.ui.activity.BaseActivity
import com.app.myapplication.ui.activity.MenuActivity
import com.app.yun.ui.MActivity
import kotlinx.android.synthetic.main.activity_login.*

class LoginActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        val pref = getSharedPreferences("pfsdata", Context.MODE_PRIVATE)
        val remeberpassword = pref.getBoolean("remeberpassword",false)
        if (remeberpassword){
            val account = pref.getString("account","")
            val password = pref.getString("password","")
            d("account",account+":"+password)
            accountEdit.setText(account)
            passwordEdit.setText(password)
            remeberCheck.isChecked = true
        }

        LoginButton.setOnClickListener{
            val account = accountEdit.text.toString()
            val password = passwordEdit.text.toString()
            val hashMap = hashMapOf("userID" to account,"password" to CommonUtil.getMD5Str(password))
            d("accountlogin",account+":"+password)
            OkHttpUtils.getInstance().post(
                this@LoginActivity,
                HttpUtil.getServiceAddress(Constant.LOGIN),
                hashMap,
                object : GsonResponseHandler<LoginData>() {
                override fun onSuccess(statusCode: Int, data: LoginData?) {
                        if (statusCode == 1) {
                            if (data != null) {
                                CookieHelper.setCookie(data.data.JSESSIONID)
                            }
                            val editor = pref.edit()
                            if (remeberCheck.isChecked){
                                editor.putBoolean("remeberpassword",true)
                                editor.putString("account",account)
                                editor.putString("password",password)
                            }else{
                                editor.clear()
                            }
                            editor.apply()                                              //提交
                            val intent = Intent(this@LoginActivity, MActivity::class.java)
                            startActivity(intent)
                        } else {
//                            "未登陆成功".showToast(this@LoginActivity)
                            Toast.makeText(this@LoginActivity,"未登陆成功",Toast.LENGTH_SHORT).show()
                        }
                       // dismissLoadingView()
                    }

                override fun onFailure(statusCode: Int, error_msg: String) {
                   // dismissLoadingView()
                    Toast.makeText(this@LoginActivity,"网络问题",Toast.LENGTH_SHORT).show()
                   // UIUtil.INSTANCE.showToast("网络问题")//login_fail_net
                }
                })

        }
    }

}
